sync/sync.go

```diff
@@ -380,7 +381,67 @@ func isRevisionCountSame(fromClient, toClient *replicaClient.ReplicaClient, disk
 	return true, nil
 }
 
+func (t *Task) initalizeSyncProgress(fromClient, toClient *replicaClient.ReplicaClient, disks []string) error {
+	rwReplica, err := fromClient.GetReplica()
+	if err != nil {
+		return err
+	}
+	woReplica, err := toClient.GetReplica()
+	if err != nil {
+		return err
+	}
+
+	snapshots := map[string]*types.Snapshot{}
+	var rwSize, woSize int64
+
+	for i := range disks {
+		disk := disks[len(disks)-1-i]
+		if _, ok := rwReplica.Disks[disk]; ok {
+			diskSize, err := strconv.ParseInt(rwReplica.Disks[disk].Size, 10, 64)
+			if err != nil {
+				return fmt.Errorf("Failed to parse size: %s of RW replica, err: %v", rwReplica.Disks[disk].Size, err)
+			}
+			rwSize += diskSize
+			snapshots[disk] = &types.Snapshot{
+				RWSize: rwReplica.Disks[disk].Size,
+				Status: types.RebuildPending,
+				WOSize: func() string {
+					if _, ok := woReplica.Disks[disk]; !ok {
+						return "NA"
+					}
+					return woReplica.Disks[disk].Size
+				}(),
+			}
+		}
+	}
+
+	for _, disk := range disks {
+		if _, ok := woReplica.Disks[disk]; ok {
+			diskSize, err := strconv.ParseInt(woReplica.Disks[disk].Size, 10, 64)
+			if err != nil {
+				return fmt.Errorf("Failed to parse size: %s of WO replica, err: %v", woReplica.Disks[disk].Size, err)
+			}
+			woSize += diskSize
+		}
+	}
+
+	syncInfo := types.SyncInfo{
+		RWReplica:            fromClient.GetAddress(),
+		WOReplica:            toClient.GetAddress(),
+		Snapshots:            snapshots,
+		RWSnapshotsTotalSize: strconv.FormatInt(rwSize, 10),
+		WOSnapshotsTotalSize: strconv.FormatInt(woSize, 10),
+	}
+	rebuild.AddSyncInfo(&syncInfo)
```

Comment:
lets add a comment that this memory is not cleared to avoid unnecessary locking code due to its access in rebuild.GetReplicaInfo